### PR TITLE
📖 Update commands.md

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR updates `commands.md` to track recent changes. There are several updates.
- more accurately reflect how commands are sensitive to non-standard environment variables and kube config more broadly;
- indicate which commands are not converted to using the space framework yet;
- change some "workspace" terminology to "space" terminology (this PR only does a little of what is needed about that).

## Related issue(s)

Fixes #

### CODE REVIEW

File: docs/content/Coding Milestones/PoC2023q1/commands.md [hunk 1]
```diff
@@ -474,18 +475,21 @@ kubestellar-version
 ## Creating SyncTarget/Location pairs
 
 In this PoC, the interface between infrastructure and workload
-management is inventory API objects.  Specifically, for each edge
-cluster there is a unique pair of SyncTarget and Location objects in a
-so-called inventory management workspace.  These kinds of objects were
+management is inventory API objects.  Specifically, for each workload
+execution cluster there is a unique pair of SyncTarget and Location
+objects in a so-called inventory space.  These kinds of objects were
```
Comment (ezrasilvera) [inline docs/content/Coding Milestones/PoC2023q1/commands.md:480]: The line below (about kcp) is not important, should we remove it ?

File: docs/content/Coding Milestones/PoC2023q1/commands.md [hunk 2]
```diff
@@ -500,12 +504,22 @@ and have at least the given labels.
 The flags can also appear anywhere later on the command line.
 
 The acceptable flags include all those of `kubectl` except for
-`--context`.  This command also accepts the following flags.
+`--context`.  BUT do not actually supply any of these flags; this
+command handles them incorrectly.
```
Comment (ezrasilvera) [inline docs/content/Coding Milestones/PoC2023q1/commands.md:508]: I wouldn't put the reason here, it's either supported or not. If it's  not maybe: "currently --context is not supported" ?

File: docs/content/Coding Milestones/PoC2023q1/commands.md (hunk 1 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/commands.md:480]: kcp TMC still exists and some people know that KubeStellar has history with TMC, so I think it is best to leave this disclaimer in for a while. I think that once the message of "kcp decoupling" is well established that should demolish any lingering expectation of TMC usage and then we can remove this disclaimer.

File: docs/content/Coding Milestones/PoC2023q1/commands.md (hunk 2 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/commands.md:508]: I would not worry about word-smithing this too carefully, it is being removed in #1444 .

File: docs/content/Coding Milestones/PoC2023q1/commands.md [hunk 3]
```diff
@@ -608,24 +605,38 @@ KUBECONFIG=$imw_space_config kubectl kubestellar remove location demo1
...
+- `-X`: turn on debug echoing of the script as it executes.
+- `--space-name $space_name`: the name of the space.
+- `--output $kubeconfig_pathname`: where to write the kubeconfig
+- `--provider-name $provider`: the name of the space provider to use;
+  defaults to `default`.
+- `--sm-core-config $SM_CONFIG`: pathname of the kubeconfig file for
+  access to the SMAS.
+- `--sm-context $SM_CONTEXT`: the name of the context to use in that
+  kubeconfig for access to the SMAS.
+- `--in-cluster`: when the core components are running in a hosting
```
Comment (ezrasilvera) [inline docs/content/Coding Milestones/PoC2023q1/commands.md:631]: I think a simpler explanation might be "when the caller resides in the same hosting cluster as the space-provider"

File: docs/content/Coding Milestones/PoC2023q1/commands.md [hunk 4]
```diff
@@ -666,6 +680,12 @@ The acceptable flags include all those of `kubectl` except for
 - `-X`: turn on debug echoing of commands inside the script that
   implements this command.
 
+This command is sensitive to the followig environment variables.
```
Comment (ezrasilvera) [inline docs/content/Coding Milestones/PoC2023q1/commands.md:683]: "sensitive" is not clear here:  maybe "this command relays on the SM_CONFIG and SM_CONTEXT variables. Make sure they are set before running the command" ? Or something like that ..

File: docs/content/Coding Milestones/PoC2023q1/commands.md [hunk 5]
```diff
@@ -742,8 +762,9 @@ the list of things that went into the WEC.
 
 The following command is a combination of `kubectl kubestellar
 ensure-location` and `kubectl kubestellar prep-for-syncer`, and takes
-the union of their command line flags and arguments.  Upon completion,
-the kcp current workspace will be what it was at the start.
+the union of their command line flags and arguments and reads the
+union of their environment variable read sets.  Upon completion, the
+kcp current workspace will be what it was at the start.
```
Comment (ezrasilvera) [inline docs/content/Coding Milestones/PoC2023q1/commands.md:767]: This is KCP only thing. We no longer have the notion of navigation. There is no "current" space

File: docs/content/Coding Milestones/PoC2023q1/commands.md (hunk 3 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/commands.md:631]: Remember that contributors can deploy the core stuff as bare processes, so it is possible that there is no hosting cluster.

File: docs/content/Coding Milestones/PoC2023q1/commands.md (hunk 4 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/commands.md]: fixed

File: docs/content/Coding Milestones/PoC2023q1/commands.md (hunk 5 above)
Comment (MikeSpreitzer) [reply docs/content/Coding Milestones/PoC2023q1/commands.md]: fixed

**Pull request merged**